tensorboard/main.py

```diff
@@ -31,6 +31,11 @@
 from tensorboard import program
 
 
+def run():
```

Comment:
IMO it's confusing to have both run() and main() with slightly different semantics.  Do we expect there to be a lot of custom TensorBoards out there that are already calling `main.main` from their own code (like `main.create_tb_app` and the other deprecated ones below)?

I'd favor moving towards having just a single `main()` function here that basically does

```python
def main():
  def bound_main(unused_argv=None):
    return program.main(default.get_plugins(), default.get_assets_zip_provider())
  tf.app.run(bound_main)
```

Then basically users can choose to either use TensorBoard's existing `main()` function here exactly as-is, or copy it and tweak as needed, e.g. if they wanted to execute their own `tf.app.run()` for some reason.

Right now there are three separate entry points that users could choose from - `program.main()` just does the core work, `main.main()` adds in the default plugins and assets, and `main.run()` adds in `tf.app.run()` (in reality there's even an earlier level where users don't even use the premade `program.main()`, like internally).  That seems a bit more complicated than we really need - I think we should aim for just a "do everything" one (here) and a "lets you customize plugins and assets" one (`program.main`).

Reply:
My goal with this change is only to quick fix a bug and I'm hesitant to change API behavior. For example, we might need the two functions here because the user might have a wrapper script where app.run() was already called.

Reply:
Ok, fair enough, but then could we at least rename run() to run_main() so that it's obvious that it just calls main()?  Otherwise it's unclear what the precedence is (does run() call main()? does main() call run()? are they totally unrelated?).

Reply:
FWIW I would still prefer to see run() renamed to run_main() in this PR before merging, but I'll approve for now and if not addressed here hopefully we can clarify the name in a follow-up.